kyo-core/shared/src/test/scala/kyo/StreamCoreExtensionsTest.scala

```diff
@@ -0,0 +1,109 @@
+package kyo
+
+class StreamCoreExtensionsTest extends Test:
+
+    "factory" - {
+        "collectAll" in run {
+            Choice.run {
+                for
+                    size <- Choice.get(Seq(0, 1, 32, 100))
+                    s1     = Stream.init(0 to 99 by 3)
+                    s2     = Stream.init(1 to 99 by 3)
+                    s3     = Stream.init(2 to 99 by 3)
+                    merged = Stream.collectAll(Seq(s1, s2, s3), size)
+                    res <- merged.run
+                yield assert(res.sorted == (0 to 99))
+            }.andThen(succeed)
+        }
+
+        "collectAllHalting" in runNotJS {
```

Comment:
@fwbrasil this is the test that's failing in JS. I don't understand why this one fails, but `mergeHaltingRight/Left` does not.